### PR TITLE
Fwmt 1382 using redis to track case ids for hh messages

### PR DESCRIPTION
**Pull request by SteveScorfield**

### CODE REVIEW

File: src/main/java/uk/gov/ons/census/fwmt/rmadapter/service/impl/JobCacheManagerImpl.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+@Service
+public class JobCacheManagerImpl implements JobCacheManager {
+
+  @Autowired
+  private RedisUtil<HouseholdRequestEntity> redisUtil;
+
+  @Override
+  public HouseholdRequestEntity cacheCreateHouseholdRequest(HouseholdRequestEntity householdRequestEntity) {
+    redisUtil.putValue(householdRequestEntity.getCaseId(), householdRequestEntity);
+    log.info("Placed the following in cache: " +  householdRequestEntity.toString());
```
Comment (sdiaz1000) [inline src/main/java/uk/gov/ons/census/fwmt/rmadapter/service/impl/JobCacheManagerImpl.java]: lets not log this

File: src/main/java/uk/gov/ons/census/fwmt/rmadapter/service/impl/JobCacheManagerImpl.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+@Slf4j
+@Service
+public class JobCacheManagerImpl implements JobCacheManager {
+
+  @Autowired
+  private RedisUtil<HouseholdRequestEntity> redisUtil;
+
+  @Override
+  public HouseholdRequestEntity cacheCreateHouseholdRequest(HouseholdRequestEntity householdRequestEntity) {
```
Comment (sdiaz1000) [inline src/main/java/uk/gov/ons/census/fwmt/rmadapter/service/impl/JobCacheManagerImpl.java]: move this code to HouseholdStore

File: src/main/java/uk/gov/ons/census/fwmt/rmadapter/service/impl/RMAdapterServiceImpl.java [hunk 3]
```diff
@@ -37,26 +39,43 @@ public void sendJobRequest(ActionInstruction actionInstruction) throws GatewayEx
...
-    jobServiceProducer.sendMessage(CanonicalJobHelper.newUpdateJob(actionInstruction));
-    gatewayEventManager.triggerEvent(actionInstruction.getActionUpdate().getCaseId(), CANONICAL_UPDATE_SENT);
+    if (householdStore.retrieveCache(actionInstruction.getActionUpdate().getCaseId()) != null) {
+      if (actionInstruction.getActionUpdate().getAddressType().equals("HH")) {
+        jobServiceProducer.sendMessage(CanonicalJobHelper.newUpdateJob(actionInstruction));
+        gatewayEventManager.triggerEvent(actionInstruction.getActionUpdate().getCaseId(), CANONICAL_UPDATE_SENT);
+      } else {
+        throw new GatewayException(GatewayException.Fault.SYSTEM_ERROR, "Valid address type not found");
+      }
+    } else {
```
Comment (sdiaz1000) [inline src/main/java/uk/gov/ons/census/fwmt/rmadapter/service/impl/RMAdapterServiceImpl.java:54]: remove the else, let them die silently, actually the surrounding if statement could read
if (householdStore.retrieveCache(actionInstruction.getActionUpdate().getCaseId()) == null)  return;

File: src/main/java/uk/gov/ons/census/fwmt/rmadapter/service/impl/RMAdapterServiceImpl.java [hunk 4]
```diff
@@ -37,26 +39,43 @@ public void sendJobRequest(ActionInstruction actionInstruction) throws GatewayEx
...
+      throw new GatewayException(GatewayException.Fault.SYSTEM_ERROR, "No matching case ID was found for " +
+              actionInstruction.getActionUpdate().getCaseId() + " when a Action Update Request was tempted.");
+    }
   }
 
   private void createCancelMessage(ActionInstruction actionInstruction) throws GatewayException {
-    if (actionInstruction.getActionCancel().getAddressType().equals("HH")) {
-      jobServiceProducer.sendMessage(CanonicalJobHelper.newCancelJob(actionInstruction));
-      gatewayEventManager.triggerEvent(actionInstruction.getActionCancel().getCaseId(), CANONICAL_CANCEL_SENT);
+    if (householdStore.retrieveCache(actionInstruction.getActionCancel().getCaseId()) != null) {
```
Comment (sdiaz1000) [inline src/main/java/uk/gov/ons/census/fwmt/rmadapter/service/impl/RMAdapterServiceImpl.java:62]: See comment above

**Review by sdiaz1000 — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/ons/census/fwmt/rmadapter/service/impl/JobCacheManagerImpl.java (hunk 1 above)
Comment (SteveScorfield) [reply src/main/java/uk/gov/ons/census/fwmt/rmadapter/service/impl/JobCacheManagerImpl.java]: Logging removed.

File: src/main/java/uk/gov/ons/census/fwmt/rmadapter/service/impl/JobCacheManagerImpl.java (hunk 2 above)
Comment (SteveScorfield) [reply src/main/java/uk/gov/ons/census/fwmt/rmadapter/service/impl/JobCacheManagerImpl.java]: Moved the cache into the HouseholdStore and removed the JobCacheManager and Impl

File: src/main/java/uk/gov/ons/census/fwmt/rmadapter/service/impl/RMAdapterServiceImpl.java (hunk 3 above)
Comment (SteveScorfield) [reply src/main/java/uk/gov/ons/census/fwmt/rmadapter/service/impl/RMAdapterServiceImpl.java:54]: As discussed, this has been altered.

File: src/main/java/uk/gov/ons/census/fwmt/rmadapter/service/impl/RMAdapterServiceImpl.java (hunk 4 above)
Comment (SteveScorfield) [reply src/main/java/uk/gov/ons/census/fwmt/rmadapter/service/impl/RMAdapterServiceImpl.java:62]: As above

**Review by sdiaz1000 — APPROVED** [review]

**Pull request merged**